### PR TITLE
test(node/fs): enable `test-fs-write-file.js`

### PR DESCRIPTION
**Pull request by F3n67u**
part of: #911

### CODE REVIEW

File: node/_tools/suites/parallel/test-fs-write-file.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+{
+  // Test read-only mode
+  const filename = join(tmpdir.path, 'test6.txt');
+  fs.writeFileSync(filename, '');
+
+  // TODO: Correct the error type
+  const expectedError = common.isWindows ? /EPERM/ : /EBADF/;
+  fs.writeFile(filename, s, { flag: 'r' }, common.expectsError(expectedError));
```
Comment (F3n67u) [inline node/_tools/suites/parallel/test-fs-write-file.js:113]: This test will fail, I don't know how to fix it. Need help! How can I convert the error from `Deno.remove` to node style error?

```
error: Uncaught (in promise) AssertionError: The input did not match the regular expression /EBADF/. Input:

'Error: Bad file descriptor (os error 9)'

    Error.captureStackTrace(this, stackStartFn || stackStartFunction);
          ^
    at new AssertionError (file:///Users/feng/dev/deno_std/node/assertion_error.ts:530:11)
    at createAssertionError (file:///Users/feng/dev/deno_std/node/assert.ts:26:17)
    at validateThrownError (file:///Users/feng/dev/deno_std/node/assert.ts:756:11)
    at Function.throws (file:///Users/feng/dev/deno_std/node/assert.ts:136:7)
    at file:///Users/feng/dev/deno_std/node/_tools/suites/common/index.js:94:12
    at file:///Users/feng/dev/deno_std/node/_tools/suites/common/index.js:194:15
    at file:///Users/feng/dev/deno_std/node/_fs/_fs_writeFile.ts:70:7
```

Comment (kt3k) [reply node/_tools/suites/parallel/test-fs-write-file.js:113]: I think the error is from fs.writeFile (of std/node), and we should map Deno error to Node error around here: https://github.com/denoland/deno_std/blob/d14d829b44e130fbe4267629ce685c69d42b7d57/node/_fs/_fs_writeFile.ts#L63

Node.js error classes are available in https://github.com/denoland/deno_std/blob/main/node/_errors.ts

Comment (F3n67u) [reply node/_tools/suites/parallel/test-fs-write-file.js:113]: fixed, but seem like ci is broken at another unrelated error.

**Comment by F3n67u** [comment]
blocked by ci broken issue: https://github.com/denoland/deno_std/issues/1589.

**Comment by F3n67u** [comment]
> blocked by ci broken issue: #1589.

ci issue has been fixed, ready for review @kt3k

**Review by kt3k — APPROVED** [review]
@F3n67u LGTM. Nice work! Thank you for your contribution!

**Comment by F3n67u** [comment]
> @F3n67u LGTM. Nice work! Thank you for your contribution!

@kt3k  thanks for your review.

**Pull request merged**